backend/auth/handlers_user_account_test.go

```diff
@@ -0,0 +1,489 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright 2024 Canonical Ltd.
+
+package auth
+
+import (
+	"errors"
+	"fmt"
+	"net/http"
+	"net/http/httptest"
+	"strings"
+	"testing"
+
+	"github.com/gin-gonic/gin"
+	"github.com/omec-project/webconsole/dbadapter"
+	"go.mongodb.org/mongo-driver/bson"
+	"golang.org/x/crypto/bcrypt"
+)
+
+type MockMongoClientEmptyDB struct {
+	dbadapter.DBInterface
+}
+
+type MockMongoClientDBError struct {
+	dbadapter.DBInterface
+}
+
+type MockMongoClientInvalidUser struct {
+	dbadapter.DBInterface
+}
+
+type MockMongoClientSuccess struct {
+	dbadapter.DBInterface
+}
+
+type MockMongoClientRegularUser struct {
+	dbadapter.DBInterface
+}
+
+type MockMongoClientDuplicateCreation struct {
+	dbadapter.DBInterface
+}
+
+func hashPassword(password string) string {
+	hashed, err := bcrypt.GenerateFromPassword([]byte(password), bcrypt.DefaultCost)
+	if err != nil {
+		return ""
+	}
+	return string(hashed)
+}
+
+func (m *MockMongoClientEmptyDB) RestfulAPIGetOne(collName string, filter bson.M) (map[string]interface{}, error) {
```

Comment:
```suggestion
func (m *MockMongoClientEmptyDB) RestfulAPIGetOne(collName string, filter bson.M) (map[string]interface{}, error) {
```
We need to use the same receiver for the same interface. Here we use `m` or ` db`. We should use one of them as a receiver in all methods.